target/linux/octeon/base-files/lib/upgrade/platform.sh

```diff
@@ -1,116 +1,110 @@
-
-		echo "${rootfsdev}"
-	fi
+platform_octeon_copy_config() {
+	local device="$1"
+	[ -n "$device}" ] && [ -b "$device" ] && {
+		mount -t "vfat" "$device" "/mnt"
+		cp -af "$UPGRADE_BACKUP" "/mnt/$BACKUP_FILE"
+		umount "/mnt"
+	}
 }
 
 platform_copy_config() {
 	case "$(board_name)" in
-	erlite)
-		mount -t vfat /dev/sda1 /mnt
-		cp -af "$UPGRADE_BACKUP" "/mnt/$BACKUP_FILE"
-		umount /mnt
-		;;
-	itus,shield-router)
-		mount -t vfat /dev/mmcblk1p1 /mnt
-		cp -af "$UPGRADE_BACKUP" "/mnt/$BACKUP_FILE"
-		umount /mnt
-		;;
+		erlite)
+			platform_octeon_copy_config "/dev/sda1"
+			;;
+		itus,shield-router)
+			platform_octeon_copy_config "/dev/mmcblk1p1"
+			;;
 	esac
 }
 
 platform_do_flash() {
-	local tar_file=$1
-	local board=$2
-	local kernel=$3
-	local rootfs=$4
-
-	local board_dir=$(tar tf "$tar_file" | grep -m 1 '^sysupgrade-.*/$')
-	board_dir=${board_dir%/}
-	[ -n "$board_dir" ] || return 1
-
-	mkdir -p /boot
-
-	if [ $board = "itus,shield-router" ]; then
-		# mmcblk1p1 (fat) contains all ELF-bin images for the Shield
-		mount /dev/mmcblk1p1 /boot
-
-		echo "flashing Itus Kernel to /boot/$kernel (/dev/mmblk1p1)"
-		tar -Oxf $tar_file "$board_dir/kernel" > /boot/$kernel
-	else
-		mount -t vfat /dev/$kernel /boot
-
-		[ -f /boot/vmlinux.64 -a ! -L /boot/vmlinux.64 ] && {
-			mv /boot/vmlinux.64 /boot/vmlinux.64.previous
-			mv /boot/vmlinux.64.md5 /boot/vmlinux.64.md5.previous
-		}
+	local sysupgrade_file="$1"
+	local kernel_file="$2"
+	local kernel_checksum_file="$kernel_file.md5"
+	local boot_device="$3"
+	local root_device="$4"
+	# find a sysupgrade directory by looking inside sysupgrade file
+	local sysupgrade_directory="$(tar tf $sysupgrade_file | awk -F '/' '/^sysupgrade/ {print $1; exit}')"
+	[ -n "$sysupgrade_directory" ] || return 1
+
+	if [ ! -f "$sysupgrade_file" ] && \
+		[ ! -b "$boot_device" ] && \
+		[ ! -b "$root_device" ] ; then
+		return 1
+	fi
 
-		echo "flashing kernel to /dev/$kernel"
-		tar xf $tar_file $board_dir/kernel -O > /boot/vmlinux.64
-		md5sum /boot/vmlinux.64 | cut -f1 -d " " > /boot/vmlinux.64.md5
+	local boot_directory="/boot"
+	mkdir -p "$boot_directory"
+	mount -t "vfat" "$boot_device" "$boot_directory"
+	# backups current kernel file to a kernel file with .previous postfix
+	if [ -f "$boot_directory/$kernel_file" ] && \
+		[ ! -L "$boot_directory/$kernel_file" ] && \
+		[ -f "$boot_directory/$kernel_checksum_file" ] && \
+		[ ! -L "$boot_directory/$kernel_checksum_file" ] ; then
+		mv "$kernel_file" "$kernel_file.previous"
+		mv "$kernel_checksum_file" "$kernel_checksum_file.previous"
 	fi
 
-	echo "flashing rootfs to ${rootfs}"
-	tar xf $tar_file $board_dir/root -O | dd of="${rootfs}" bs=4096
+	echo "copying kernel to $boot_directory/$kernel_file"
+	tar xf "$sysupgrade_file" "$board_dir/kernel" -O > "$kernel_file"
+	md5sum "$kernel_file" | cut -f1 -d " " > "$kernel_checksum_file"
+
+	echo "flashing rootfs to $root_device"
+	tar xf "$sysupgrade_file" "$board_dir/root" -O | dd of="$root_device" bs=4096
```

Comment:
same here...

Reply:
sorry, just pushed changes resolving that issue.
thank you